src/components/ProductItem.jsx

```diff
@@ -1,34 +1,27 @@
 import { Link, useNavigate } from "react-router-dom";
 import { useAuth, useProducts } from "../contexts";
+import { addToCart, addToWishlist, checkLogin, removeFromWishlist } from "../utils";
 
 const ProductItem = ({product}) => {
 
-    const {imageUrl,isLiked,title,quantity,price,rating} = product;
-    const {isLoggedIn, userDetails,setUserDetails} = useAuth();
-    const {wishList,cartList} = userDetails;
+    const {imageUrl,title,quantity,price,rating} = product;
+    const {userDetails,dispatchUser} = useAuth();
+    const {isLoggedIn,wishList,cartList} = userDetails;
     const navigate = useNavigate();
 
-    const addToCart=(product)=>{
-        if(isLoggedIn){
-            (cartList.find((item)=>item._id===product._id))?
-             setUserDetails({...userDetails, cartList: cartList.map((item)=>(item._id===product._id)?{...item, count: item.count+1}:item)})
-            :
-            setUserDetails({...userDetails, cartList: [...cartList, {...product, count: 1}]})
-        }else{
-            navigate("/login")
+    const handleCart=(product)=>{
+        if(checkLogin(isLoggedIn)){
+            addToCart(product, isLoggedIn, dispatchUser);   
         }
     }
 
-    const {allProducts, setAllProducts} = useProducts();
-
-    const addToWishlist = (product) => {
-        if(isLoggedIn){
-            (wishList.find((item)=>item._id===product._id))
-            ?console.log("Item already exists"):
-            setUserDetails({...userDetails, wishList: [...wishList, product]})
-        }
-        else{
-        navigate("/login")
+    const handleWishlist = (product) => {
+        if(checkLogin(isLoggedIn)){
+            wishList.find((item)=>item._id===product._id)
```

Comment:
you can extract this outside of this function like `isAddedToWishlist = wishlist.find(item=>item._id===product._id)` since you may need this variable to change the look of the wishlist button on a card accordingly

Reply:
Yes. Will do this. Thanks